README.md

```diff
@@ -1 +1,163 @@
-# opencraft
+OpenCraft
+=========
+
+Install
+-------
+
+Instructions based on Ubuntu 14.04.
+
+Install the system package dependencies & virtualenv:
+
+```
+$ sudo apt-get install `cat debian_packages.lst`
+$ pip3 install --user virtualenv && pip3 install --user virtualenvwrapper
+```
+
+Ensure you load virtualenv with Python 3 in `~/.bashrc`:
+
+```
+export PATH="$PATH:$HOME/.local/bin" VIRTUALENVWRAPPER_PYTHON=/usr/bin/python3
+source $HOME/.local/bin/virtualenvwrapper.sh
+```
+
+Then reload `~/.bashrc`, create the virtual env and install the Python requirements:
+
+```
+$ . ~/.bashrc
+$ mkvirtualenv -p /usr/bin/python3 opencraft
+$ workon opencraft
+$ pip install -r requirements.txt
+```
+
+
+Configure
+---------
+
+Create an `.env` file at the root of the repository or set environment variables, customizing the
+settings from `opencraft/settings.py` which are loaded via `env()`.
+
+
+Run
+---
+
+To run the development server:
+
+```
+$ make rundev
+```
+
+Then go to:
+
+* User interface: [http://localhost:5000/](http://localhost:5000/)
+* API: [http://localhost:5000/api/](http://localhost:5000/api/)
+* Admin: [http://localhost:5000/admin/](http://localhost:5000/admin/)
+
+To run the production server:
+
+```
+$ make run
+```
+
+To change the number of concurrent workers ran by the production server:
+
+```
+$ make run WORKERS=2
+```
+
+
+Process description
+-------------------
+
+This runs three processus via honcho, which reads `Procfile` or `Procfile.dev` and loads the
+environment from the `.env` file:
+
+* *web*: the main HTTP server (Django - Werkzeug debugger in dev, gunicorn in prod)
+* *websocket*: the websocket server (Tornado)
+* *worker*: runs asynchronous jobs (Huey)
+
+Important: the Werkzeug debugger started by the development server allows remote execution
+of Python commands. It should *not* be run in production.
+
+
+Static assets collection
+------------------------
+
+The Web server started in the development environment also doesn't require to run collectstatic
+after each change.
+
+The production environment automatically runs collectstatic on startup, but you can also run it
+manually:
+
+```
+$ make collectstatic
+```
+
+
+Migrations
+----------
+
+Similarly, migrations are run automatically on startup - for both the development and production
```

Comment:
I'm not sure I like that... Is this a common setup in python/django world? It makes it very easy to mess up your database if you accidentally check out/deploy a wrong branch. Ideally it would print a warning to let you know you have pending migrations, but not run them automatically.